trivial-builders: replace writeReferencesToFile with writeClosure (#178717)
NixOS/nixpkgs: doc/build-helpers/trivial-build-helpers.chapter.md

## `writeReferencesToFile` {#trivial-builder-writeReferencesToFile}

Deprecated. Use [`writeReferenceClosureToFile`](#trivial-builder-writeReferenceClosureToFile) instead.

> ## `writeReferenceClosureToFile` {#trivial-builder-writeReferenceClosureToFile}

Reviewer: Also, `writeReferencesClosureToFile` is a bit of a mouthful, I can imagine myself typoing or forgetting if references are plural

> Writes the closure of transitive dependencies of the given paths to a file.

Reviewer: Nit: to me this almost reads like we're given "paths" which are "paths to a file", rather than "we write to a file".

Just so I propose at least something: "given a list of store paths (???), writes the closure of their transitive dependencies to a file".

Author: It's directly taken from the documentation of `writeReferencesToFile`, with "the given path" -> "the given paths".

The new version sounds more clearly.

Author: I modify the doc. Please take a look.

> Given a list of [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) (or string-like expressions coercible to store paths), it writes the combined [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) of them to a text file.

Reviewer: ```suggestion
Builds a line-separated file with all of the [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) in the [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) of a list of root paths
```

Reviewer: Could also make it

```suggestion
Builds a line-separated file with [references](https://nixos.org/manual/nix/stable/glossary#gloss-reference) to all of the [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) in the [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) of a list of root paths
```

Idk

Reviewer: "root paths" is kind of informal though

Reviewer: @fricklerhandwerk sorry to ping but we're discussing names and documentation, would you like to share an opinion?

Author: Thank you for helping!

"References of all store paths in the closure" is equivalent to "the closure". I think it might be confusing to mention *reference* here.

Reviewer 2: ```suggestion
Given a list of [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) (or string-like expressions coercible to store paths), write their  collective [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) to a text file.
```

I (strongly) recommend using imperative in reference documentation, primarily for brevity. "Do this, do that."

@SomeoneSerge the original structure was actually better because the prose matches the functional interface: given <input> make <output>.

@ShamrockLee [to combine](https://en.wiktionary.org/wiki/combine#Etymology) means "to put together two things". Since the input is a list, the word should be along the lines of [to compose](https://en.wiktionary.org/wiki/compose#Etymology) or [to collect](https://en.wiktionary.org/wiki/collect#Etymology_1).

Reviewer: Yes, "given x, do y" is actually my preference too; what I intended with the suggestion is to specify the format ("line-separated") and get rid of the "combined closure".

On the latter note, I'd just drop "combined": "the closure of a set of paths" is already meaningful, there's no need for extra adjectives. The extra adjectives can also be harmful if they are perceived as technical terms which they are not
